homeassistant/components/swiss_public_transport/coordinator.py

```diff
@@ -67,12 +67,15 @@ def __init__(self, hass: HomeAssistant, opendata: OpendataTransport) -> None:
         )
         self._opendata = opendata
 
-    def remaining_time(self, departure) -> timedelta | None:
+    def remaining_time(self, departure) -> str | None:
         """Calculate the remaining time for the departure."""
         departure_datetime = dt_util.parse_datetime(departure)
 
         if departure_datetime:
-            return departure_datetime - dt_util.as_local(dt_util.utcnow())
+            try:
```

Comment:
I would rather prefer a check ´departure_datetime < now()´ instead of using a try block. Seems cleaner and better understandable to me.

Reply:
I agree that the intent would be cleaner, but then there's a race condition no? Some time will pass between the two `now()` calls (one here, one inside `get_time_remaining`), and if we are unlucky it's enough to cause an exception.  
I might be nitpicking here so I'm happy to change it if we think this is fine.

Reply:
I agree, good argument. I would prefer the cleaner style but ultimately the home assistant team will decide😊